Lib/pathlib.py

```diff
@@ -734,27 +643,38 @@ def is_absolute(self):
         a drive)."""
         if not self._root:
             return False
-        return not self._flavour.has_drv or bool(self._drv)
+        return self._flavour is posixpath or bool(self._drv)
 
     def is_reserved(self):
         """Return True if the path contains one of the special names reserved
         by the system, if any."""
-        return self._flavour.is_reserved(self._parts)
+        if self._flavour is posixpath or not self._parts:
```

Comment:
Is it good to hard-code a check for a specific flavour?

Reply:
I'd like to do this

```python
return self._flavour.isreserved(self)
```

But it would require a new `os.path.isreserved()` function - see #88569. What do you think?

Reply:
I've created a PR that adds `os.path.isreserved()`: #95486.

Reply:
@brettcannon FYI, Eryk Sun and Steve Dower are helping refine the `os.path.isreserved()` implementation in #95486. It will take a while to get right! As such I'm going to push a commit to _this_ PR that adds `os.path._isreserved()` (note the underscore). Hopefully unblocks this PR! I'll fix up merge conflicts from `main` when they happen.